scanner/ssh_scanner.go

```diff
@@ -58,7 +58,7 @@ func (s SSHScanner) ScanProtocol(conn net.Conn, host *Target, timeout time.Durat
 	}
 
 	var sshResult SSHResult
-	if serverInfo != nil {
+	if serverInfo != nil && serverInfo.ServerVersion != "" && serverInfo.Key != nil {
```

Comment:
Why do we need to check for `serverInfo.Key`?

Reply:
it would cause a panic if the key is nil

Reply:
The question is *why*? Which function causes the panic? Are there circumstances where we might not get a key but still want to store the other information?

Reply:
Fingerprinting the key happens a couple of lines below that check, if the key is nil this functions panics immediately
I supposed that if we cannot gather the key itself we won't get any other information because how should we establish a connection if the server doesn't provide a key.

Reply:
Can we write an empty string into the fingerprint if `serverInfo.Key` is nil?

Reply:
Yes, will do